### PR TITLE
adding the userInfo state to the NGRX

### PR DESCRIPTION
**Pull request by udaydenduluri33**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-2967

### Change description ###

Get rid of the Roles cookie
Use NGRX state for storing Roles information
Lastly, get rid of all the old references of cookie usage

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/app/components/get-help/get-help.component.spec.ts [hunk 1]
```diff
@@ -64,13 +86,13 @@ describe('GetHelpComponent', () => {
     });
   });
 
-  describe('Verify HTML content on Get help page', () => {
+  // describe('Verify HTML content on Get help page', () => {
```
Comment (udaydenduluri33) [inline src/app/components/get-help/get-help.component.spec.ts:89]: @connorpgpmcelroy  to uncomment and fix the test

File: src/app/containers/app-header/app-header.component.ts [hunk 2]
```diff
@@ -156,15 +137,15 @@ export class AppHeaderComponent implements OnInit, OnDestroy {
...
-
-    this.featureToggleService.getValue<Theme[]>('mc-application-themes', this.getDefaultApplicationThemes())
-      .subscribe(applicationThemes => {
-
-        const applicationTheme: Theme = this.getApplicationThemeForUser(applicationThemes);
-
-        this.hideNavigationListener(this.store);
-
-        this.setAppHeaderProperties(applicationTheme);
+    this.userDetails$ = this.store.pipe(select(fromActions.getUserDetails));
```
Comment (udaydenduluri33) [inline src/app/containers/app-header/app-header.component.ts:140]: @connorpgpmcelroy  to add unit test for this scenario combinelatest

File: src/app/containers/app-header/app-header.component.spec.ts [hunk 3]
```diff
@@ -21,13 +21,7 @@ const storeMock = {
...
  * Serialised roles as returned from the Node layer.
  */
-const SERIALISED_ROLES = 'j:["pui-organisation-manager", "caseworker-publiclaw", "caseworker-divorce-financialremedy-solicitor", "caseworker"]';
-
-const cookieServiceMock = {
-  get: () => {
-    return SERIALISED_ROLES;
-  },
-};
+// const SERIALISED_ROLES = 'j:["pui-organisation-manager", "caseworker-publiclaw", "caseworker-divorce-financialremedy-solicitor", "caseworker"]';
```
Comment (rayliangatos) [inline src/app/containers/app-header/app-header.component.spec.ts:24]: please tidy this up

File: src/app/guards/acceptTerms.guard.spec.ts [hunk 4]
```diff
@@ -20,18 +19,17 @@ describe('Accept terms guard', () => {
...
 
     xit('loads when not yet loaded', () => {
-        mockCookie.get.and.callFake((key) => {
-            if (key === 'roles') { return 'pui-case-manager'; }
-            if (key === '__userid__') { return 'userId'; }
-        });
+        // mockCookie.get.and.callFake((key) => {
+        //     if (key === 'roles') { return 'pui-case-manager'; }
+        //     if (key === '__userid__') { return 'userId'; }
+        // });
```
Comment (rayliangatos) [inline src/app/guards/acceptTerms.guard.spec.ts:32]: please tydy this up

File: src/app/containers/app-header/app-header.component.spec.ts (hunk 3 above)
Comment (adnanakgun) [inline src/app/containers/app-header/app-header.component.spec.ts:24]: Commented code

**Review by rayliangatos — APPROVED** [review]
approved and please tidy up those code commented out

File: src/app/guards/acceptTerms.guard.spec.ts [hunk 5]
```diff
@@ -1,17 +1,16 @@
-import { CookieService } from 'ngx-cookie';
 import { of } from 'rxjs';
 import { TermsConditionsService } from '../services/terms-and-conditions/terms-and-conditions.service';
 import { AcceptTermsGuard } from './acceptTerms.guard';
 
 describe('Accept terms guard', () => {
     let guard: AcceptTermsGuard;
     let mockStore: any;
-    let mockCookie: jasmine.SpyObj<CookieService>;
+    // let mockCookie: jasmine.SpyObj<CookieService>;
```
Comment (adnanakgun) [inline src/app/guards/acceptTerms.guard.spec.ts:8]: Commented code again

File: src/app/guards/acceptTerms.guard.spec.ts [hunk 6]
```diff
@@ -1,17 +1,16 @@
...
 
     beforeEach(() => {
         mockStore = jasmine.createSpyObj('mockStore', ['unsubscribe', 'dispatch', 'pipe']);
-        mockCookie = jasmine.createSpyObj<CookieService>('mockCookie', ['get']);
-        guard = new AcceptTermsGuard(mockStore, mockCookie);
+        // mockCookie = jasmine.createSpyObj<CookieService>('mockCookie', ['get']);
```
Comment (adnanakgun) [inline src/app/guards/acceptTerms.guard.spec.ts:12]: Commented code again

File: src/app/guards/acceptTerms.guard.spec.ts [hunk 7]
```diff
@@ -20,18 +19,17 @@ describe('Accept terms guard', () => {
     });
 
     xit('returns true for non pui users', (done: any) => {
```
Comment (adnanakgun) [inline src/app/guards/acceptTerms.guard.spec.ts:21]: About time we get rid `xit`s if we are not going to enable them.

File: src/app/guards/acceptTerms.guard.spec.ts [hunk 8]
```diff
@@ -20,18 +19,17 @@ describe('Accept terms guard', () => {
...
             done();
         });
     });
 
     xit('loads when not yet loaded', () => {
-        mockCookie.get.and.callFake((key) => {
-            if (key === 'roles') { return 'pui-case-manager'; }
-            if (key === '__userid__') { return 'userId'; }
-        });
+        // mockCookie.get.and.callFake((key) => {
```
Comment (adnanakgun) [inline src/app/guards/acceptTerms.guard.spec.ts:29]: `xit` and commented code again

**Review by adnanakgun — APPROVED** [review]

**Comment by udaydenduluri33** [comment]
Its already merged to master as a part of different JIRA.

**Pull request closed without merging**